tests/net_inet/asyncio_http_https_opc.py

```diff
@@ -0,0 +1,81 @@
+# Test simple HTTP request with asyncio.open_connection()
+
+try:
+    import os
+    import asyncio
+    import ssl
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+
+async def http_get(url):
+    reader, writer = await asyncio.open_connection(url, 80)
+
+    print("write GET")
+    writer.write(b"GET / HTTP/1.0\r\n\r\n")
+    await writer.drain()
+
+    print("read response")
+    data = await reader.read(100)
+    print("read:", data.split(b"\r\n")[0])
+
+    print("close")
+    writer.close()
+    await writer.wait_closed()
+    print("done")
+
+
+# Test simple HTTPS request with uasyncio.open_connection()
+
+
+# This certificate was obtained from micropython.org using openssl:
+# $ openssl s_client -showcerts -connect micropython.org:443 </dev/null 2>/dev/null
+# The certificate is from Let's Encrypt:
+# 1 s:/C=US/O=Let's Encrypt/CN=R3
+#   i:/C=US/O=Internet Security Research Group/CN=ISRG Root X1
+# Validity
+#            Not Before: Sep  4 00:00:00 2020 GMT
+#            Not After : Sep 15 16:00:00 2025 GMT
+# Copy PEM content to a file (certmpy.pem) and convert to DER e.g.
+# $ openssl x509 -in certmpy.pem -out certmpy.der -outform DER
+# Then convert to hex format, eg using binascii.hexlify(data).
+
+ca_cert_chain = "mpycert.der"
+
+try:
+    os.stat(ca_cert_chain)
+except OSError:
+    print("SKIP")
+    raise SystemExit
+
+client_ctx = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+client_ctx.verify_mode = ssl.CERT_REQUIRED
+client_ctx.load_verify_locations(cafile=ca_cert_chain)
```

Comment:
This test (and the other 3 added to the `net_inet` directory) don't run on CPython (at least not 3.11.6) because CPython's `load_verify_locations()` requires the file to be in PEM format, not DER.

The fix for that is to load the data from the DER here and then pass it in as `cadata`.

But, that still doesn't seem to have the correct behaviour because I would think that you don't need the line below (the call to `load_default_certs()`) because it should be enough to load just the DER certificate from `mpycert.der` in order to access micropython.org.  But if I remove the `load_default_certs()` then CPython cannot connect to micropython.org.  If I leave `load_default_certs()` in place then I can just remove this `load_verify_locations()` call and CPython works.

So it seems this test is not working as expected on CPython.

Reply:
> The fix for that is to load the data from the DER here and then pass it in as cadata.

I agree

> But, that still doesn't seem to have the correct behaviour because I would think that you don't need the line below (the call to load_default_certs()) because it should be enough to load just the DER certificate from mpycert.der in order to access micropython.org. But if I remove the load_default_certs() then CPython cannot connect to micropython.org. If I leave load_default_certs() in place then I can just remove this load_verify_locations() call and CPython works.

I could reproduce this, and it seems `mbedtls` is a bit more "permissive" than "OpenSSL" when it comes to certificate validation, i.e. `mbedtls` is OK using "intermediate" CA certificates, while "OpenSSL" wants the "ultimate" root CA certificate.

The fix is using this root CA certificate that both MicroPython and CPython accept 👍🏼 

FWIW I got this root CA certificate from https://github.com/micropython/micropython-lib/pull/633 so if you have time you may want to have a look.

Reply:
OK, using the ISRG certificate is a good idea.  Except it doesn't seem to work for me, running that test I get:


CPython:
```
ssl.SSLError: not enough data: cadata does not contain a certificate (_ssl.c:4020)
```

MicroPython:
```
ValueError: invalid cert
```